evm/src/cpu/kernel/asm/memory/packing.asm

```diff
@@ -41,12 +41,243 @@ global mload_packing_u64_LE:
 // Pre stack: context, segment, offset, value, len, retdest
 // Post stack: offset'
 global mstore_unpacking:
-    // stack: context, segment, offset, value, len, retdest
-    %stack(context, segment, offset, value, len, retdest) -> (context, segment, offset, value, len, offset, len, retdest)
-    // stack: context, segment, offset, value, len, offset, len, retdest
-    MSTORE_32BYTES
-    // stack: offset, len, retdest
-    ADD SWAP1
+    %stack(context, segment, offset, value, len, retdest) -> (len, context, segment, offset, value, retdest)
+    PUSH 3
+    //stack: BYTES_PER_JUMP, len, context, segment, offset, value, retdest
+    MUL
+    //stack: jump_offset, context, segment, offset, value, retdest
+    PUSH mstore_unpacking_0
+    //stack: mstore_unpacking_0, jump_offset, context, segment, offset, value, retdest
+    ADD
+    //stack: address_unpacking, context, segment, offset, value, retdest
+    JUMP
+
+mstore_unpacking_0:
```

Comment:
Are we sure this will properly panic? Because otherwise it will yield an unverifiable proofs, as the Bytepacking table cannot handle empty segments, which would fail the CTL verification.

Reply:
additionally I'd add indentation to the %rep below for clarity, as is usually done.
```asm
    %rep 3
        PUSH 0
    %endrep
```

Reply:
I believe you're never supposed to call `mstore_unpacking` with length `0`, but I might be wrong. @Nashtare?

Reply:
We're never supposed, but I don't think there is such handling as I mentioned above. So we may as well add this logic here, by skipping completely the read/write if len==0?
For now it is only enforced through `memcpy`/`memset`.